middleware/administration/unittest/source/cli/test_queue.cpp

```diff
@@ -490,5 +490,66 @@ b2        external  50233  out    B
          }
       }
 
+      TEST( cli_queue, clear)
+      {
+         common::unittest::Trace trace;
+
+         auto domain = local::domain( R"(
+domain:
+   queue:
+      groups:
+         -  alias: Q
+            queues:
+               -  name: a
+)");
+
+         // enqueue 10 messages
+         {
+            auto capture = local::execute( R"(echo "casual" | casual buffer --compose | casual buffer --duplicate 10 | casual queue --enqueue a | casual pipe --human-sink | wc -l)");
+            EXPECT_TRUE( std::stoi( capture.standard.out) == 10) << CASUAL_NAMED_VALUE( capture);
+         }
+
+         // clear
+         {
+            auto capture = local::execute( R"(casual queue --clear a --porcelain true)");
+            constexpr std::string_view expected = R"(a|10
+)";
+            EXPECT_TRUE( capture.standard.out == expected) << CASUAL_NAMED_VALUE( capture);
+         }
+
+         // verify queue is empty
+         {
+            auto capture = local::execute( R"(casual queue --list-messages a --porcelain true)");
+            EXPECT_TRUE( capture.standard.out.empty()) << CASUAL_NAMED_VALUE( capture);
+         }
+      }
+
+      TEST( cli_queue, remove_messages)
```

Comment:
I wanted to write one with `--force` but couldnt figure out how to get some messages in the right state, any ideas?

Reply:
Hmm, It's going to be a bit cumbersome...
* create a queue-group with a specific queuebase name.
* enqueue some messages
* shutdown the queue group (or the domain)
* open the queuebase and alter the state of the messages 
* boot the queue group
* perform tests

Reply:
Right, thats not too bad, I'll have a go at it